core/src/main/scala/io/finch/EndpointCompiled.scala

```diff
@@ -0,0 +1,37 @@
+package io.finch
+
+import cats.effect.{ConcurrentEffect, Effect, IO}
+import com.twitter.finagle.Service
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.{Future, Promise}
+
+/**
+  * A compiled endpoint that is represented as a function of `Request => F[(Trace, Response)]`
+  * with ability to convert it to `com.twitter.finagle.Service`
+  */
+abstract class EndpointCompiled[F[_]] extends (Request => F[(Trace, Response)]) { self =>
```

Comment:
What do you think about `EndpointCompiler[F[_]]` being just an alias for a [Kleisli](https://typelevel.org/cats/datatypes/kleisli.html)? This would allow us to reuse most of the Cats' goodness w/o needing to introduce a `Filter` abstraction.

Reply:
Yea, it's very much possible

Reply:
Except, I don't know where to put `toService` conversion then

Reply:
I believe we can just define it for `Kleisli[F[_], Request, Response]`.

Reply:
Not really.

I had to work around it with `Endpoint.run` method

Reply:
I would prefer not to have implicits inside of wildcard package that is imported everywhere, but choices are limited here.

Reply:
I agree. Let's have it as `CompileOps` for now but keep brainstorming.